shim/src/interpose.c

```diff
@@ -54,10 +54,10 @@
 
 #define INTERPOSE_CALL(type, fn_libc, fn_demi, ...)           \
     {                                                         \
-        bool reentrant = is_reentrant_demi_call();            \
-                                                              \
         init_libc();                                          \
```

Comment:
What is the rationale for that?

Reply:
I need to initialize the reentrancy guards hashset before calling is_reentrant_demi_call. The initialization for that is in init_libc. I'll change the name of that function to be more descriptive, since now it is not initializing only libc.